tools/get-upload-stats.py

```diff
@@ -0,0 +1,580 @@
+            )
+        s += self.commit_delta.to_markdown()
+        if self.since_latest is not None and self.since_latest:
+            s += "\n\n" + self.since_latest.to_markdown()
+        return s
+
+
+class DandiDataSet(BaseModel):
+    path: Path
+
+    def readgit(self, *args: Union[str, Path], **kwargs: Any) -> str:
+        return cast(
+            str,
+            subprocess.check_output(["git", *args], cwd=self.path, text=True, **kwargs),
+        ).strip()
+
+    def get_first_and_last_commit(
+        self, from_dt: Optional[datetime], to_dt: Optional[datetime]
+    ) -> Tuple[CommitInfo, CommitInfo]:
+        # --before orders by commit date, not author date, so we need to filter
+        # commits ourselves.
+        cmtlines = self.readgit(
+            "log",
+            r"--grep=\[backups2datalad\]",
+            r"--grep=Ran backups2datalad\.py",
+            "--format=%H %aI %p",
+        ).splitlines()
+        commits: List[CommitInfo] = []
+        warned_nonlinear = False
+        for cmt in cmtlines:
+            committish, created, *parents = cmt.strip().split()
+            if len(parents) > 1 and not warned_nonlinear:
+                log.warning("Commits in given timeframe are nonlinear")
+                warned_nonlinear = True
+            short_id = self.readgit("describe", "--always", committish)
+            commits.append(
+                CommitInfo(committish=committish, short_id=short_id, created=created)
+            )
+        commits.sort(key=attrgetter("created"))
+        commits = filter_commits(commits, from_dt, to_dt)
+        if not commits:
+            raise ValueError("No commits in given timeframe")
+        elif len(commits) == 1:
+            raise ValueError("Only one commit in given timeframe")
+        return (commits[0], commits[-1])
+
+    def get_tags(
+        self, from_dt: Optional[datetime], to_dt: Optional[datetime]
+    ) -> List[CommitInfo]:
+        # Tags are returned in ascending order of creation
+        taglines = self.readgit(
+            "tag",
+            "-l",
+            "--sort=creatordate",
+            "--format=%(creatordate:iso-strict) %(refname:strip=2)",
+        ).splitlines()
+        tags: List[CommitInfo] = []
+        for tl in taglines:
+            ts, _, tag = tl.partition(" ")
+            tags.append(CommitInfo(committish=tag, short_id=tag, created=ts))
+        return filter_commits(tags, from_dt, to_dt)
+
+    def get_assets(self, commit: CommitInfo) -> Iterator[AssetInfo]:
+        repo = AnnexRepo(str(self.path))
+        for p, info in repo.get_content_annexinfo(ref=commit.committish).items():
+            relpath = p.relative_to(self.path)
+            if info.get("type") == "file" and relpath.parts[0] not in IGNORED:
+                path = str(PurePosixPath(relpath))
+                subject: Optional[str]
+                if m := re.fullmatch(r"sub-([^/]+)/[^/]+\.nwb", path):
```

Comment:
I don't think we should limit to .nwb files here (we have .tif etc now), so I guess smth like 
```suggestion
                if m := re.match(r"sub-([^/]+)/[^/]+", path):
```
?

@satra WDYT -- is it worth "ignoring" .json side car files? I would say - we should not bother (i.e. we should count them since they are also assets ATM and likely in the future)

Reply:
@yarikoptic Note that the limiting to NWB files only applies to extracting the subject from the asset path; aside from calculating the subject, the script still takes into account all files other than `.dandi/`, `.datalad/`, `.gitattributes`, and `dandiset.yaml`.

Reply:
But we have dandisets with subjects and without .nwb files - I still don't see how it is nwb specific